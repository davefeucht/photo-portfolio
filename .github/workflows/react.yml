# This workflow will do a clean install of node dependencies, build the source code and run tests across different versions of node
# For more information see: https://help.github.com/actions/language-and-framework-guides/using-nodejs-with-github-actions

name: React CI/CD

on:
  push:
    branches: [ master ]
  pull_request:
    branches: [ master ]

jobs:
  build:

    runs-on: ubuntu-latest

    strategy:
      matrix:
        node-version: [18.x]

    steps:
    - name: Checkout code
      uses: actions/checkout@v2

    - name: Install Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}

    - name: Enable corepack
      run: corepack enable

    - name: Install dependencies
      run: yarn install

    - name: Build project
      run: npm run build --if-present

    - name: Run tests
      run: npm test
