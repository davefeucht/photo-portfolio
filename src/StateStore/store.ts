import { configure, makeAutoObservable } from 'mobx';
import { ImageData, Image, Store, Category, Post, Page } from '../utils/types';

configure({ enforceActions: 'observed' });

class stateStore implements Store {
    menuState = 'closed'
    applicationRoot: HTMLElement = null;
    screenInfo: {
        width: number,
        height: number
    }
    siteInfo: {
        siteName: string,
        siteUrl: string
    }
    visiblePost: {
        postId: number,
        postTitle: string,
        tags: number[],
        tagNames: string[],
        fullImageUrl: string,
        width: number,
        height: number
    };
    categoryList: Category[] = [];
    currentCategoryPosts: Post[] = [];
    currentCategoryData: Category = {
        id: 0,
        count: 0,
        description: '',
        link: '',
        name: '',
        slug: '',
        taxonomy: '',
        parent: 0
    };
    pages: Page[] = [];
    currentPageData: Page = {
        id: 0,
        link: '',
        modified: '',
        slug: '',
        type: '',
        title: {
            rendered: ''
        },
        content: {
            rendered: '',
            protected: false
        },
        author: 0,
        featured_media: 0
    };
    maxItemsPerPage = 10;

    constructor() {
        makeAutoObservable(this);
    }

    setMenuState = (state: string) => {
        this.menuState = state;
    }

    toggleMenuState = () => {
        this.menuState = this.menuState === 'closed' ? 'open' : 'closed';
    }

    setApplicationRoot = (element: HTMLElement) => {
        this.applicationRoot = element;
    }

    setSiteName = (name: string) => {
        this.siteInfo.siteName = name;
    }

    setVisiblePost = (postId: number, postTitle: string) => {
        this.visiblePost.postId = postId;
        this.visiblePost.postTitle = postTitle;
    }

    setVisiblePostImage = (fullImageUrl: string) => {
        this.visiblePost.fullImageUrl = fullImageUrl;
    }

    setVisiblePostTags = (tagNames: string[]) => {
        tagNames.forEach(tagName => {
            this.visiblePost.tagNames.push(tagName);
        });
    }

    clearVisiblePostTagNames = () => {
        this.visiblePost.tagNames.splice(0, this.visiblePost.tagNames.length);
    }

    setCategoryList = (categories: Category[]) => {
        this.categoryList.length = 0;
        categories.forEach((category, index) => {
            this.categoryList[index] = category;
        })
    }

    setCategoryPosts = (posts: Post[]) => {
        this.currentCategoryPosts.length = 0;
        posts.forEach((post, index) => {
            this.currentCategoryPosts[index] = post;
        })
    }

    setCategoryData = (categoryData: Category) => {
        Object.keys(categoryData).forEach((key: keyof Category) => {
            this.currentCategoryData[key] = categoryData[key];
        });
    }

    setThumbnailImageUrl = (imageData: ImageData) => {
        this.currentCategoryPosts[imageData.post_index].thumbnail_image = imageData.image_url;
    }

    setCurrentPost = (postData: Post) => {
        Object.keys(postData).forEach((key: keyof Image) => {
            this.visiblePost[key] = postData[key];
        })
    }

    setPages = (pages: Page[]) => {
        pages.forEach(page => {
            this.pages.push(page);
        });
    }

    setPageData = (pageData: Page) => {
        Object.assign(this.currentPageData, pageData);
    }
}

export default stateStore;
