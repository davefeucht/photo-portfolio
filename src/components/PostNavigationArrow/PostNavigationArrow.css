.post-navigation-arrow {
  position: absolute;
  height: 80px;
  width: 40px;
  border: 1px solid #000000;
  background-color: rgba(128, 128, 128, 0.4);
  transition: opacity 300ms ease-in-out;
  opacity: 1;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  cursor: pointer;
}

.post-navigation-arrow.active {
  opacity: 1;
  cursor: pointer;
}

.post-navigation-arrow.disabled {
  opacity: 0.5;
  cursor: default;
}

.post-navigation-arrow > .arrow {
  position: absolute;
  color: #FFFFFF;
  font-size: initial;
}

.arrow > img {
  vertical-align: middle;
}