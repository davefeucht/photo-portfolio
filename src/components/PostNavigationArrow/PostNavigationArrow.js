/****************
* PostNavigationArrow component displays a navigation arrow for a Post
****************/

import React from "react";
import { observer } from 'mobx-react';
import { Link, useParams } from 'react-router-dom';
import './PostNavigationArrow.css';

const PostNavigationArrow = observer(({ stateStore, direction, api }) => {
  const { categoryId } = useParams();

  const currentIndex = stateStore.currentCategoryPosts.findIndex(post => post.id === stateStore.visiblePost.postId);
  const numberOfPosts = stateStore.currentCategoryPosts.length;
  let nextIndex = direction === 'previous' ? currentIndex - 1 : currentIndex + 1;
  if (nextIndex < 0) {
    nextIndex = numberOfPosts - 1;
  }
  if (nextIndex > numberOfPosts - 1) {
    nextIndex = 0;
  }
  const post = stateStore.currentCategoryPosts[nextIndex];

  //TODO: calc this based on a height variable, not just a hard-coded 41
  const top = stateStore.visiblePost.height ? (stateStore.visiblePost.height / 2) - (41) : 0;
  const divStyle = direction === 'previous' ? { top: `${top}px`, left: '0px' } : { top: `${top}px`, right: '0px' };
  const imgSrc = direction === 'previous' ? './images/arrow-left.png' : './images/arrow-right.png';

  return(
    <div className="post-navigation-arrow" style={divStyle}>
      <div className="arrow"><img src={imgSrc} /></div>
    </div>
  );
  /*
  return(
    <div className="post-navigation-arrow" style={divStyle}>
      <Link to={`/category/${categoryId}/posts/${post.id}`}>
        <div className="arrow"><img src={imgSrc} /></div>
      </Link>
    </div>
  );
  */
});

PostNavigationArrow.displayName = 'PostNavigationArrow';

export default PostNavigationArrow;