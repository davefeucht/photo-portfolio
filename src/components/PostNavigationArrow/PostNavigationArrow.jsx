/** **************
* PostNavigationArrow component displays a navigation arrow for a Post
*************** */

import './PostNavigationArrow.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React from 'react';
import { Link, useParams } from 'react-router-dom';

const PostNavigationArrow = ({ stateStore, direction, postId }) => {
    const { categoryId } = useParams();

    // TODO: calc this based on a height variable, not just a hard-coded 41
    const top = stateStore.visiblePost.height ? (stateStore.visiblePost.height / 2) - (41) : 0;
    const divStyle = direction === 'previous' ? { top: `${top}px`, left: '0px' } : { top: `${top}px`, right: '0px' };
    const classNames = ['post-navigation-arrow', postId ? 'active' : 'disabled', direction];

    const getContent = () => {
        if (postId) {
            return (
                <Link to={`/category/${categoryId}/post/${postId}`}>
                    <div className="arrow"><img src="./assets/images/chevron.svg" /></div>
                </Link>
            );
        }
        return (
            <div className="arrow"><img src="./assets/images/chevron.svg" /></div>
        );
    };

    return (
        <div className={classNames.join(' ')} style={divStyle}>
            {getContent()}
        </div>
    );
};

PostNavigationArrow.displayName = 'PostNavigationArrow';

PostNavigationArrow.propTypes = {
    stateStore: PropTypes.object.isRequired,
    direction: PropTypes.string.isRequired,
    postId: PropTypes.number
};

PostNavigationArrow.defaultProps = {
    postId: null
};

export default observer(PostNavigationArrow);
