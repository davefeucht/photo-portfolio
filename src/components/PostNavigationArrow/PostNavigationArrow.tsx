/** **************
* PostNavigationArrow component displays a navigation arrow for a Post
*************** */

import './PostNavigationArrow.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { Link, useParams } from 'react-router-dom';

interface PostNavigationArrowProps {
    imageHeight: number,
    direction: string,
    postId: number
}

const PostNavigationArrow: React.FC<PostNavigationArrowProps> = ({ imageHeight = 0, direction, postId }) => {
    const { categoryId } = useParams();

    // TODO: calculate these based on a variable somewhere - they are navigationArrowHeight - (postHeaderHight + postFooterHeight)
    // What we want here is actually the height of just the image and not the whole modal
    const top = (imageHeight / 2) - 41;
    const divStyle = direction === 'previous' ? { top: `${top}px`, left: '0px' } : { top: `${top}px`, right: '0px' };
    const classNames = ['post-navigation-arrow', postId ? 'active' : 'disabled', direction];

    const getContent = () => {
        if (postId) {
            return (
                <Link to={`/category/${categoryId}/post/${postId}`}>
                    <div className="arrow"><img src="./assets/images/chevron.svg" /></div>
                </Link>
            );
        }
        return (
            <div className="arrow"><img src="./assets/images/chevron.svg" /></div>
        );
    };

    return (
        <div className={classNames.join(' ')} style={divStyle}>
            {getContent()}
        </div>
    );
};

PostNavigationArrow.displayName = 'PostNavigationArrow';

export default observer(PostNavigationArrow);
