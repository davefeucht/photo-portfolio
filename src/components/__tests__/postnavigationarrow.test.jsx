import { render } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';

import StateStore from '../../StateStore/store';
import PostNavigationArrow from '../PostNavigationArrow/PostNavigationArrow';

const direction = 'previous';
const postId = 35;

test('PostNavigationArrow displays', () => {
    const store = new StateStore();
    const { container } = render(
        <MemoryRouter>
            <PostNavigationArrow stateStore={store} direction={direction} postId={postId} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
});
