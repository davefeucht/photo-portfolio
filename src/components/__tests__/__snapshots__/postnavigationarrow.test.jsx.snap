// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostNavigationArrow displays 1`] = `
<div
  class="post-navigation-arrow active previous"
  style="top: 0px; left: 0px;"
>
  <a
    href="/category/undefined/post/35"
  >
    <div
      class="arrow"
    >
      <img
        src="./assets/images/chevron.svg"
      />
    </div>
  </a>
</div>
`;
